### ISSUE TITLE
Teleports are broken

### ISSUE THREAD

**Issue by Trs5405** [MAFINS/MenyooSP#423]
Teleports are broken. Sometimes work, but very rarely

**Comment by mistahwolf** [comment]
Yes, I reported that a while ago with no response. As is, if you're in a vehicle and you've selected 'vehicle access' on the marker you can enter and exit with no prob. On foot...forget it. They just dropped another update so hopefully it's fixed....

**Comment by Trs5405** [comment]
> Yes, I reported that a while ago with no response. As is, if you're in a vehicle and you've selected 'vehicle access' on the marker you can enter and exit with no prob. On foot...forget it. They just dropped another update so hopefully it's fixed....

Where did you see they dropped an update? Im seeing most recent 14 days ago

**Comment by mistahwolf** [comment]
1.7.2 Was the one I mentioned. I was using 7.1. Either way, after testing the portals issue hasn't been addressed. You can still get in with a vehicle, but not on foot.

**Comment by itsjustcurtis** [comment]
Can you both elaborate on this? I am unable to replicate this issue. Using teleport options on foot seem to work fine.

**Comment by mistahwolf** [comment]
@itsjustcurtis , sorry just saw this today. I'm not sure what to tell you. Teleports had just stopped working when entering on foot. Just before this last update, during a session it worked 'once'. I thought it might have been interference from the 'Teleportals' mod I was testing for entering ymaps. When I took it out, it was still broken. I just downloaded 1.7.3 and will see if it still has the same issue.

**Comment by mistahwolf** [comment]
@itsjustcurtis Yep, just checked out 1.7.3 and teleports are still broken. They do not respond to player on foot.

**Comment by itsjustcurtis** [comment]
I need you to elaborate further on "they don't respond". I am using them in my gMe and they work fine. If I'm unable to replicate the issue I can't hope to fix it. What exactly do you do to get the unresponsiveness? Which menu options do you select. Where is your character, what is it doing, have you set a different skin etc....

**Comment by mistahwolf** [comment]
@itsjustcurtis, there's not much more to tell. 'Unresponsive' means what I said previously - 'When in a vehicle, teleports function properly. When on foot, they don't respond.' As I've mentioned in my earlier thread, 'whether as one of the main 3 characters' or 'in any other skin, when on foot my avatar walks into the teleport marker and nothing happens.' All my builds that require teleports has this problem. To 'get unresponsiveness' I place my character in close proximity to the marker, step into it or pass through it...and...nothing...happens. This didn't start happening until after the second Contract update. Have you updated your game and not regressed it to an earlier version? Others have mentioned this issue. I am using the retail/PC version with the 'R' version launcher. It's confusing that Menyoo's teleports work when in a vehicle but not on foot. It's like the code for selecting whether a vehicle is allowed access to the teleport has been switched whereas before you could walk through as the default.

**Comment by Trs5405** [comment]
@itsjustcurtis exactly like he said...vehicle teleports work fine. But when you load a Menyoo spooner map that has teleports only for on foot use, they do not work. Whether its one of the 3 main characters, a ped or MP model, the teleportation feature does not work on foot. You walk into the circle or whatever you've designated as your marker, and nothing happens. The game doesnt try to load you anywhere...absolutely nothing happens when walking into a teleport marker on foot.

**Comment by itsjustcurtis** [comment]
Thank you both, thats what I needed. I was using the teleport menu, which works fine. I was not aware of any markers in Spooner mode. I should be able to check properly and hopefully find a solution

**Comment by itsjustcurtis** [comment]
Can either of you send me an example of a spooner map with the problem? I've done some testing and am unable to replicate using the markers in spooner mode; Teleports seem to work fine both on foot and in vehicle for me.

**Comment by mistahwolf** [comment]
@itsjustcurtis here's a map I released you can test. The markers are invisible, but you can change their opacity as needed. There are both foot and vehicle markers on the doors and garage.
https://www.gta5-mods.com/maps/senora-desert-garage

**Comment by itsjustcurtis** [comment]
@mistahwolf I've got this in-game and appears to be working just fine for me. Both the blue pedestrian door and the large garage door are teleporting me inside just fine. Not sure anything can be done with menyoo to fix the issue you're both having :/

**Comment by mistahwolf** [comment]
@itsjustcurtis Uhy! Thanks for looking into this. So to narrow down what could be causing this again I ask;
1) Are you using an updated version 'Contract  second update' or later? This update is when the problem started.
2) Are you using a Steam, Retail or 'R' downloaded version of the game?
3) Are you testing this w/o Mods (OIV enabled)?
4) Something is definitely interfering with the markers code as it makes zero sense that vehicles can go through with a ped in it, but peds on foot can't. Any ideas what that could be? 
I ask these questions before pulling my Mods folder and going through everything including what's in the scripts folder.

**Comment by itsjustcurtis** [comment]
Happy to answer questions

1) 2802.0, drug wars
2) Steam
3) No, menyoo won't work with mods disabled (unless you are referring just to stuff in the mods folder, in which case nothing here should have an effect anyway)
4) Struggling for ideas, the first place i would be looking is at scripts, they're the most likely thing to be interfering with anything.

**Comment by mistahwolf** [comment]
@itsjustcurtis, yeah I'm thinking it may be a script as well. It definitely isn't the 'Teleportals' mod because I pulled it and the issue was still there. And yes I meant 'No Mods Folder' and it didn't seem to make sense an OIV enabled mod should be messing with this.

@Trs5405, I like HKH's work but don't use it because it's not 'flexible'. I pick various locations to put teleports for the vanilla interiors and they're never where he puts them. Also all his stuff is strictly for the  main 3 characters which I never use. And yes, I will let you know if I figure this out. I need to as it's a PIA....

**Comment by Trs5405** [comment]
@mistahwolf I will continue testing as well and see what i can come up with! I will update you soon...hopefully!

**Comment by mistahwolf** [comment]
@Trs5405 I'm still using plain Open Interiors. I used to have the 'enhanced version', but I like to place MP interiors in different areas as that better suits the scenarios I'm working. It still works just fine and I don't have to fool with using the in-game phone to do things.

**Comment by mistahwolf** [comment]
@itsjustcurtis @Trs5405 , Finally figured it out. The culprit was 'No Boundary Limits'. Specifically, the 'Max Depth Limit' in the .ini has to be set at 'true' or it breaks the teleports in Menyoo. The reason I had it set to 'false' was because it caused glitches with the player ped not being able to enter the sea w/o teleporting back to shore and it also caused a glitch that won't allow you to teleport into any of the interiors below the docks where Simeon's secret garage is. Markers (so far) seem to work again. Try this and let me know if you have the same issues seeing/fixing this glitch.

**Comment by itsjustcurtis** [comment]
Ahh I wonder if setting that to true disables the native that Menyoo relies
on to teleport. If that is the case then I'm not sure what can be done on
our side to fix.

Kind regards

Curtis Elliott

**Comment by Trs5405** [comment]
@mistahwolf Thats very interesting, I've been using no bounday limits for a LONG time and have never made any INI changes to it. Something must've changed in Menyoo to conflict with that..

Great work none the less my friend, this is huge!!

**Comment by Trs5405** [comment]
@mistahwolf Hmmm thats interesting, Ive been using no boundary limits for a long time, and have never made an y changes in the ini, something in menyoo must've changed. Great work none the less! 

In using open interiors, how do you get it to include the new MP DLC interiors?

**Comment by Trs5405** [comment]
@mistahwolf I tried removing no boundary limits all together, and the teleports still arent working. I just placed a few down to try out and its still broken for me, even after removing no boundary limits all together. 

Also, do you have a version of open interiors that includes the newer MP DLC maps?

**Comment by mistahwolf** [comment]
@itsjustcurtis Yeah, I'm not sure either as the glitch started occurring when the NBL team added the ability for submersibles to go below the 'hard floor' 'R' put in to keep them from going too deep. I've contacted them several times but they are MIA....
Now that seems to be 'fixed', if you have time/interest, could I get you to take a look at another map I'm having trouble getting to load? In the meantime, thanks for your input. Most modders rarely take the time to answer questions or even look for potential fixes!

@Trs5405 Yes, first I removed NBL to see if that was the issue and once confirmed, I suspected the depth limit was the main cause. I reset the depth in the .ini to 350 (I've had it deep as 999). No, I am using the original 'Open Interiors' as many of the currently useable interiors are empty which makes less hassle building new ymaps inside them. Sounds like you have the newer versions HKH made and he has buttloads of scripts attached to those. Try pulling those out and then test NBL. From what Curtis mentioned 'it's a native thing' and in your situation you're getting double blocked.

**Comment by Trs5405** [comment]
@mistahwolf Im very perplexed by this issue. Ive always used EAI by HKH in conjunction with menyoo and havent had issues until recently - (last 4 months). I will try removing EAI by HKH, however i dont think that is the issue because ive used them both for years with no issue. 

I will do some testing and get back to you.

**Comment by mistahwolf** [comment]
@Trs5405 Yeah, it all started with the $#!%%y contract dlc. When they added that garbage 'pay-to-play' content and blocked it from PC, all kinds of problems kicked in for mods that have worked for ages. So I just ran another test to see if this setup was still working and I've been finding more interiors that glitch 'cause they're under the map. I also ran the previous test with a skin over spawned franklin and got the results mentioned. The test I just ran was with a soul-stolen freemode ped so I could beat the MP Clothes glitch that crashes the game when you use menyoo to change outfits. Once more vehicles pass through markers just fine...peds on foot...fuggeddaboutit! Now, I did get 'surfaced' after teleporting into those spaces with both 'Simple Trainer' and Menyoo with MP Maps enabled. I'll try again with just the ped after a clean reload/restart.

**Comment by mistahwolf** [comment]
So yes 'No Boundary Limits' is the culprit. So if you have the 'Max Depth Limit patch' set for 'true', you can pass through markers on foot with a main character ped, a reskinned MCP or one that's been 'soul swapped.' The glitch comes in when you try to teleport into interiors that are under the map. Especially for those under the Docks where 'Simeon's Garage' is and if you try to go down into the sewers. If you select 'False' in the NBL .ini, you can't access teleport markers on foot and will even get 'surfaced' whether using the marker...or teleport directly. This happens with Menyoo, the 'Portals' mod and with the teleports in 'Open Interiors'. Any other mod using teleport markers probably has the same issue.

**Comment by mistahwolf** [comment]
@itsjustcurtis, @Trs5405  - Did some more digging and fount the 'NBL Log'. The 'ultimate culprit' for the glitches with Menyoo marker teleports comes down to the 'patch_entity_task_boundary' command. In the log it consistently fails. Selecting 'false' for that in the .ini 'fixes' things in that you can now teleport with Menyoo and other marker teleport mods on foot. However (you should have known that was coming), it's not 'perfect'. Though you 'can' use marker teleports, there are still some spots you'll still get 'surfaced'. You can get into a lot more 'undermap' interiors, including those located under the 'Simeon's Garage' docks. But, the longer your game session the more likely you'll get surfaced. Curtis, maybe there's a way to 'insulate' menyoo's native you mentioned from this. Pls try this with NBL in with and without the suggested 'fixes' to see. Alloc8r's MIA, so I don't see this getting fixed anytime soon on their end.

**Comment by Trs5405** [comment]
@mistahwolf I did a clean install, put in all the  mods i was using before except NBL and Im still having an issue with the TPs. When I spawn a MP the first time it works, but after that if i try another one and to try spawn the same one again, the TPs dont work...Im not certain NBL is the issue....

**Comment by mistahwolf** [comment]
@Trs5405, I'm inclined to agree. Further testing does show NBL does cause problems with teleports, but I'm back to menyoo being primary having an issue with teleports. I believe it's a 'memory issue' as I've noticed too that early in a game session you can get TP's to work on foot. But as the session goes on, that eventually fails. Strange enough, driving a vehicle through still doesn't have any issues. @itsjustcurtis mentioned being able to use markers, but we haven't heard whether he still could after multiple uses. Like I've said from the begining, the 'contract dlc' broke this important part of Menyoo....

**Comment by TheMajesticNoob** [comment]
Hi there!
I sadly wish to confirm that marker teleport issue continues. As of v1.8.1 we still have this problem. When you first start the game, teleport markers are working just fine. As soon as you get in and out of a car, teleport markers stops working.
I really hope you may find a solution for this issue.
Thank you in advance and have a fine day.
